apps/web/src/lib/functions/file-loaders/epub/generate-epub-html.ts

```diff
@@ -139,32 +139,41 @@ export default function generateEpubHtml(
       htmlHref = itemIdToHtmlRef[itemIdRef];
     }
 
-    const regexResult = /.*<body(?:[^>]*id="(?<id>.+?)")*[^>]*>(?<body>(.|\s)+)<\/body>.*/.exec(
-      data[htmlHref] as string
-    )!;
+    const contentParser = new DOMParser();
+    const parsedContent = contentParser.parseFromString(data[htmlHref] as string, 'text/html');
 
-    const bodyId = regexResult?.groups?.id || '';
-    let innerHtml = regexResult?.groups?.body || '';
+    const htmlClass = parsedContent.querySelector('html')?.className || '';
+
+    const body = parsedContent.querySelector('body');
+    const bodyId = body?.id || '';
+    const bodyClass = body?.className || '';
+    let innerHtml = body?.innerHTML || '';
```

Comment:
your approach can lead to empty body tags for books with bad formatted html (which the regex would not care about). workaround is to check if the body child node count length is not 0 and otherwise to reparse the html string as text/xml (and if still no child nodes are present to error out and prevent the import)